### PR TITLE
Added buttons for stage phases

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
image

image

image

### CODE REVIEW

File: src/components/Stages/index.tsx [hunk 1]
```diff
@@ -151,9 +168,61 @@ function Stage() {
...
+                  </div>
+                ) : null}
+                {showWithdrawFundsBtn ? (
+                  <div
+                    className={`${style.stageDonateNow} ${
+                      withdrawFundsBtnDisable ? style.disableBtn : ""
+                    }`}
+                    onClick={withdrawTokens}
+                  >
+                    Withdraw Funds
```
Comment (TBomberman) [inline src/components/Stages/index.tsx]: Rename to Withdraw Tokens

**Comment by TBomberman** [comment]
User should only be able to see Withdraw Tokens when the stage is complete but it shows "In progress". Is the status dynamic? There should be "In progress", "Voting and evaluation", "Success" or "Fail".

**Comment by TBomberman** [comment]
Need to show current block near by the expiration block. Need to also translate that into user's machine time in brackets.

**Comment by UsamaSaleem14** [comment]
> User should only be able to see Withdraw Tokens when the stage is complete but it shows "In progress". Is the status dynamic? There should be "In progress", "Voting and evaluation", "Success" or "Fail".

The status are dynamic. To add a screenshot I manipulated the buttons otherwise the buttons will be shown based on the status and the state of the stage.

File: src/utils/ContractUtils.ts [hunk 2]
```diff
@@ -77,3 +77,21 @@ export const fetchCurrentBlock = async (): Promise<number> => {
   const blockNumber = await provider.getBlockNumber();
   return blockNumber;
 };
+
+export const getDateFromBlockNumber = async (
+  blockNumber: number,
```
Comment (TBomberman) [inline src/utils/ContractUtils.ts]: rename this to currentBlockNumber

File: src/utils/ContractUtils.ts [hunk 3]
```diff
@@ -77,3 +77,21 @@ export const fetchCurrentBlock = async (): Promise<number> => {
...
+  expiryBlockNumber: number
```
Comment (TBomberman) [inline src/utils/ContractUtils.ts]: rename this to targetBlockNumber

**Pull request merged**